### PR TITLE
Bug 1574663 - convert to use the new community deployment

### PR DESCRIPTION
**Pull request by djmitche**
This adjusts
 * worker pools
 * codecov secret name
 * index routes

To succeed in .taskcluster.yml, this will need the https://github.com/apps/community-tc-integration app added to the repo (and the old one removed)

### CODE REVIEW

File: cinnabar/cmd/download.py [hunk 1]
```diff
@@ -94,8 +94,8 @@ def download(args):
                   'version of git cinnabar.',
                   file=sys.stderr)
             return 1
-        url = 'https://index.taskcluster.net/v1/task/github'
-        url += '.glandium.git-cinnabar.helper.'
+        url = 'https://community-tc.services.mozilla.com/api/index/v1/task/'
+        url += 'project.git-cinnabar.git-cinnabar.glandium.helper.'
```
Comment (glandium) [inline cinnabar/cmd/download.py]: There seems to be a `git-cinnabar` too many

File: .taskcluster.yml [hunk 2]
```diff
@@ -35,20 +35,20 @@ tasks:
...
         taskGroupId: ${taskGroupId}
         schedulerId: taskcluster-github
-        provisionerId: aws-provisioner-v1
-        workerType: github-worker
+        provisionerId: proj-git-cinnabar
+        workerType: ci
         retries: 5
         scopes:
           $flatten:
             - assume:repo:github.com/${e.base_login}/${e.base_repo_name}:decision-task
```
Comment (glandium) [inline .taskcluster.yml]: Is that still valid?

File: .taskcluster.yml [hunk 3]
```diff
@@ -35,20 +35,20 @@ tasks:
...
             - $if: 'tasks_for == "github-push"'
               then:
-              - 'queue:route:index.github.${e.base_login}.${e.base_repo_name}.*'
-              - secrets:get:repo:github.com/${e.base_login}.${e.base_repo_name}:codecov
+              - 'queue:route:index.project.git-cinnabar.${e.base_login}.*'
```
Comment (glandium) [inline .taskcluster.yml]: The reason the index key contained the login before is that IIRC that's the prefix I had permission to write to (which is fair, considering how the prefix was). But now that git-cinnabar appears first, and considering only pushes to the canonical repo are supposed to be indexed, I feel like the login part should go away here.

File: .taskcluster.yml [hunk 4]
```diff
@@ -35,20 +35,20 @@ tasks:
...
+              - secrets:get:project/git-cinnabar/codecov
         routes:
           $if: 'e.decision_id'
           then:
-            - 'index.github.${e.login}.${e.repo_name}.decision.${e.decision_id}'
+            - 'index.project.git-cinnabar.${e.login}.decision.${e.decision_id}'
```
Comment (glandium) [inline .taskcluster.yml]: but not here.

File: CI/start-worker.sh [hunk 5]
```diff
@@ -27,13 +27,14 @@ cat > worker.config <<EOF
...
   "tasksDir": "tasks",
+  "rootUrl": "https://community-tc.services.mozilla.com",
   "workerGroup": "proj-git-cinnabar",
   "workerId": "travis-$TRAVIS_BUILD_ID",
   "workerType": "$workerType"
 }
 EOF
 
-curl -OL https://github.com/taskcluster/generic-worker/releases/download/v10.11.0/generic-worker-darwin-amd64
+curl -OL https://github.com/taskcluster/generic-worker/releases/download/v11.1.1/generic-worker-darwin-amd64
```
Comment (glandium) [inline CI/start-worker.sh]: Could you send the generic worker update to a separate PR, I'd like to try and merge that before the switch.

Comment (djmitche) [reply CI/start-worker.sh]: #232

File: .taskcluster.yml (hunk 2 above)
Comment (djmitche) [reply .taskcluster.yml]: It is -- it's added in https://github.com/mozilla/community-tc-config/pull/37

File: .taskcluster.yml (hunk 4 above)
Comment (djmitche) [reply .taskcluster.yml]: if "not" refers to 'the login part should go away here", then that would cause tags to fail, I think -- the scope and the route would be different.  So, what do you mean?

**Comment by djmitche** [comment]
Can you spell out how you'd like the routes to look?

File: .taskcluster.yml (hunk 4 above)
Comment (glandium) [reply .taskcluster.yml]: I meant that, but you're right. And `decision_id` is only set under the same condition as the scopes, so it's actually fine to remove the login here too.

**Comment by glandium** [comment]
> Submitting the task to Taskcluster failed. Details

This is not surprising, given that I haven't removed the old taskcluster github app. I'm leaving it on purpose for now. But I also didn't enable the new one before pushing to the try branch, so I amended the head (changing its commit date), and pushed again... and nothing seems to be happening with the new taskcluster app. https://github.com/glandium/git-cinnabar/commits/try

I also need to remember how to setup client access for the Travis-CI based workers.

**Comment by glandium** [comment]
oh, the new taskcluster app added a comment there: https://github.com/glandium/git-cinnabar/commit/576c7b049aa0fc78717db8c7e83b06d0f942a8cb#commitcomment-35703563
There's a scope problem.

**Comment by djmitche** [comment]
OK, added and applied https://github.com/mozilla/community-tc-config/pull/37/commits/f670da4648e92f73171dabeae709de7b3ab85a00

File: CI/start-worker.sh [hunk 6]
```diff
@@ -27,7 +27,7 @@ cat > worker.config <<EOF
   "requiredDiskSpaceMegabytes": 512,
   "signingKeyLocation": "worker.key",
   "tasksDir": "tasks",
-  "rootURL": "https://taskcluster.net",
+  "rootURL": "https://community-tc.services.mozilla.com",
```
Comment (glandium) [inline CI/start-worker.sh]: There is an extra whitespace at the end of the line here.

File: CI/tasks.py [hunk 7]
```diff
@@ -227,8 +227,8 @@ def __init__(self, **kwargs):
                 task['metadata'][k] = task['metadata']['name'] = v
             elif k == 'index':
                 if TC_IS_PUSH:
-                    task['routes'] = ['index.github.{}.{}.{}'.format(
-                        TC_LOGIN, TC_REPO_NAME, v)]
+                    task['routes'] = [
+                        'index.project.git-cinnabar.{}'.format(v)]
```
Comment (glandium) [inline CI/tasks.py]: There's another place that needs a corresponding change on line 70.

**Comment by glandium** [comment]
That was fast :) We're good to go with this version as soon as the problem with actions is fixed.

**Comment by djmitche** [comment]
Actions are fixed in 21.3.0 which is deployed now.  Care to give it another spin?
